src/runtime/runtime.cs

```diff
@@ -1852,87 +1916,111 @@ internal static bool PyIter_Check(IntPtr pointer)
+        {
+            using var namePtr = new StrPtr(name, Encoding.UTF8);
+            return Delegates.PyImport_ImportModule(namePtr);
+        }
 
-        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
-        internal static extern IntPtr PyImport_AddModule(string name);
+        internal static IntPtr PyImport_ReloadModule(IntPtr module) => Delegates.PyImport_ReloadModule(module);
 
-        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
-        internal static extern IntPtr PyImport_GetModuleDict();
+        
+        internal static BorrowedReference PyImport_AddModule(string name)
+        {
+            using var namePtr = new StrPtr(name, Encoding.UTF8);
+            return Delegates.PyImport_AddModule(namePtr);
+        }
 
-        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
-        internal static extern void PySys_SetArgvEx(
-            int argc,
-            [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(StrArrayMarshaler))] string[] argv,
-            int updatepath
-        );
+        internal static BorrowedReference PyImport_GetModuleDict() => Delegates.PyImport_GetModuleDict();
+
+        
+        internal static void PySys_SetArgvEx(int argc, string[] argv, int updatepath)
+        {
+            var marshaler = StrArrayMarshaler.GetInstance(null);
+            var argvPtr = marshaler.MarshalManagedToNative(argv);
+            try
+            {
+                Delegates.PySys_SetArgvEx(argc, argvPtr, updatepath);
+            }
+            finally
+            {
+                marshaler.CleanUpNativeData(argvPtr);
+            }
+        }
 
         /// <summary>
         /// Return value: Borrowed reference.
         /// Return the object name from the sys module or NULL if it does not exist, without setting an exception.
         /// </summary>
-        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
-        internal static extern BorrowedReference PySys_GetObject(string name);
 
-        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
-        internal static extern int PySys_SetObject(string name, IntPtr ob);
+        internal static BorrowedReference PySys_GetObject(string name)
+        {
+            using var namePtr = new StrPtr(name, Encoding.UTF8);
+            return Delegates.PySys_GetObject(namePtr);
+        }
+
+        internal static int PySys_SetObject(string name, BorrowedReference ob)
+        {
+            using var namePtr = new StrPtr(name, Encoding.UTF8);
+            return Delegates.PySys_SetObject(namePtr, ob);
+        }
 
 
         //====================================================================
         // Python type object API
         //====================================================================
+        static readonly delegate* unmanaged[Cdecl]<IntPtr, bool> pyType_Check;
```

Comment:
Apparently unused, is this a left-over from a previous attempt? Is this one also the reason `Runtime` is declared `unsafe`?

Reply:
`Runtime` in unsafe because any use of unmanaged function pointer requires `unsafe`. So all the methods with bodies like `=> Delegates.PyFunc_Pointer(args)`, so basically all methods.